### ISSUE TITLE
Crash due to 'Unable to find the specified file.' after upgrading world to VS 1.17.0 Pre-1

### ISSUE THREAD

**Issue by TaskaRaine** [anegostudios/VintageStory-Issues#1939]
**Game Version:** 1.17.0 Pre-1
**Platform:** Windows
**Modded:** Yes(but loaded without mods)
**SP/MP:** Singleplayer

### Description
When upgrading from Vintage Story 1.16.5 to Vintage Story 1.17.0 Pre-1, a previously created world would crash on load. Yesterday, it loaded _sometimes_. Today, it hasn't loaded correctly once.

client-main seems to indicate that some vanilla mod systems would fail to prestart.

This save was originally created with VS 1.15.1 and upgraded to VS 1.16+ and has been used for mod testing. 

Note: New single-player purely vanilla saves can be loaded just fine.
Note: This save can be run on a locally hosted server which a local client can then successfully connect to. 

### How to reproduce
1. Attempt to load 'taskas freestyle lands' in a single player instance.
2. The game should crash near the end of the loading sequence.

I've not been able to reproduce this from scratch, unfortunately.

### Logs
client-main: https://gist.github.com/TaskaRaine/d585ba7824b83d2fff8bb421c31d5314
client-debug: https://gist.github.com/TaskaRaine/0043ecf7fb8055b5a6c016f9ff2c4efc
server-main: https://gist.github.com/TaskaRaine/32410abae7d83cad2902a08fb5409936
server-debug: https://gist.github.com/TaskaRaine/c9868b7c9b9d87c5763a725f97828689

The world file is available here if needed: https://www.dropbox.com/s/qjdcsypp44i9s9q/taskas%20freestyle%20lands.vcdbs?dl=0

**Comment by copygirl** [comment]
Just before the `FileNotFoundException`, there's this error:
```
[Error] Could not find shape game:item/tool/wand for item game:magicwand
```

**Comment by TaskaRaine** [comment]
Indeed, yes. The client-debug file also indicates the above shape error as well as similar errors for other item shapes as well. To confirm that the shapes do indeed exist at the filepath, here's a screenshot: 

![ShapeFilepath](https://user-images.githubusercontent.com/53409481/173907899-71e68367-205a-44ec-acae-d57f8908ee57.png)

**Comment by Craluminum2413** [comment]
Closing due to inactivity

**Comment by genderneutralnoun** [comment]
I and someone else on the Vintage Story discord are having this same crash. Please reopen.

**Comment by radfast** [comment]
The issue is caused basically by having **too many mods**!   Try playing with not so many mods and the issue should go away.

In more detail, game startup has to wait for the block shape loading to complete (just finding and parsing all the shape files for every block including every modded block), before it can proceed with next steps (for example, tesselation of items, the first item is the magic wand.    We hard-coded a 20 second timeout for that wait: in vanilla it should be less than 1 second.   It looks like in a heavily modded game it can be more than 20 seconds for some people.

This will be resolved next release (which may be 1.18.2 stable).  I increased the timeout value to 2 minutes.  If the block shapes can't all be found in 2 minutes then probably something went wrong and the game won't work anyhow.

Until then, players can fix this simply by reducing the number of mods loaded together, especially cutting down on some mods which add huge numbers of new blocks.
